### PR TITLE
Delegate topology hint gen to CPU manager policy

### PR DESCRIPTION
**Pull request by ConnorDoyle**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Delegate topology hint gen to CPU manager policy
- The previous implementation depended on a fixed set of policies.

**Which issue(s) this PR fixes**: N/A

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
```

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/policy_static.go [hunk 1]
```diff
@@ -301,3 +301,102 @@ func (p *staticPolicy) guaranteedCPUs(pod *v1.Pod, container *v1.Container) int
...
+
+func (p *staticPolicy) GetTopologyHints(s state.State, pod v1.Pod, container v1.Container) map[string][]topologymanager.TopologyHint {
+	// If there are no CPU resources requested for this container, we do not
+	// generate any topology hints.
+	if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+		return nil
+	}
+
+	// Otherwise, attempt to generate TopologyHints for the CPUManager.
+	// TODO: Generalize this so that its applicable to future CPUManager polices.
```
Comment (klueska) [inline pkg/kubelet/cm/cpumanager/policy_static.go]: Probably dont need this TODO if we go this route

Comment (ConnorDoyle) [reply pkg/kubelet/cm/cpumanager/policy_static.go]: You're right. I tried to come up with a simplified interface for policy implementers but I'm not sure if it's possible. Even enumerating the possibilities depends on the CPU management policy in deep ways. Removing.

Comment (ConnorDoyle) [reply pkg/kubelet/cm/cpumanager/policy_static.go]: done

**Comment by klueska** [comment]
@ConnorDoyle Looks like you need a rebase after the bitmask change merged.

**Pull request merged**